### PR TITLE
adding setwritable to vtctld tablet commands

### PR DESCRIPTION
**Pull request by kristee-planetscale**

### CODE REVIEW

**Comment by kristee-planetscale** [comment]
I have some open questions on this PR: 

I need confirmation on the code and also any examples that may need to be added to match the format of the other commands. 

- https://github.com/vitessio/vitess/pull/8816 
     - OSS PR 

- https://github.com/vitessio/vitess/blob/b5005a195fa7a420b3c018c9a700df19772e8313/go/cmd/vtctldclient/command/tablets.go
     - This is where my concern for the code comes in. Should StartReplication and StopReplication be changed from "tablet alias" to "alias" in example, args and errors (if there are any errors)?

**Comment by FancyFane** [comment]
I would think it is better for us to match `<tablet_alias>` as it is the more descriptive word to use here, and it is what most of the documentation uses. 

Before we push this documentation out, I was checking and it looks like this command is not available for `vtctlclient`. There is no `SetWritable` action. 

```sh
jason@linux:~$ vtctlclient -server localhost:15999 SetWritable zone1-0000000200 true
Unknown command: SetWritable
SetWritable Error: rpc error: code = Unknown desc = unknown command
E0224 17:29:27.685594   94055 main.go:76] remote error: rpc error: code = Unknown desc = unknown command
```

```sh
jason@linux:~$ vtctlclient -server localhost:15999 help
Available commands:

Tablets:
  (DEPRECATED) InitTablet [-allow_update] [-allow_different_shard] [-allow_master_override] [-parent] [-db_name_override=<db name>] [-hostname=<hostname>] [-mysql_port=<port>] [-port=<port>] [-grpc_port=<port>] [-tags=tag1:value1,tag2:value2] -keyspace=<keyspace> -shard=<shard> <tablet alias> <tablet type>
  GetTablet <tablet alias>
  (DEPRECATED) UpdateTabletAddrs [-hostname <hostname>] [-ip-addr <ip addr>] [-mysql-port <mysql port>] [-vt-port <vt port>] [-grpc-port <grpc port>] <tablet alias> 
  DeleteTablet [-allow_primary] <tablet alias> ...
  SetReadOnly <tablet alias>
  SetReadWrite <tablet alias>
  StartReplication <table alias>
  StopReplication <tablet alias>
  ChangeTabletType [-dry-run] <tablet alias> <tablet type>
  Ping <tablet alias>
  RefreshState <tablet alias>
  RefreshStateByShard [-cells=c1,c2,...] <keyspace/shard>
  RunHealthCheck <tablet alias>
  (DEPRECATED) IgnoreHealthError <tablet alias> <ignore regexp>
  Sleep <tablet alias> <duration>
  ExecuteHook <tablet alias> <hook name> [<param1=value1> <param2=value2> ...]
  ExecuteFetchAsApp [-max_rows=10000] [-json] [-use_pool] <tablet alias> <sql command>
  ExecuteFetchAsDba [-max_rows=10000] [-disable_binlogs] [-json] <tablet alias> <sql command>
  VReplicationExec [-json] <tablet alias> <sql command>
  Backup [-concurrency=4] [-allow_primary=false] <tablet alias>
  RestoreFromBackup [-backup_timestamp=yyyy-MM-dd.HHmmss] <tablet alias>
  ReparentTablet <tablet alias>
```

File: content/en/docs/13.0/reference/programs/vtctl/tablets.md [hunk 1]
```diff
@@ -232,6 +232,30 @@ Changes the db type for the specified tablet, if possible. This command is used
...
+Sets the specified tablet as writable or read-only.
+
+#### Example
+
+<pre class="command-example">SetWritable &lt;alias&gt; &lt;true/false&gt;</pre>
+
+#### Arguments
+
+* <code>&lt;tablet alias&gt;</code> &ndash; Required. A Tablet Alias uniquely identifies a vttablet. The argument value is in the format <code>&lt;cell name&gt;-&lt;uid&gt;</code>.
+* <code>&lt;tablet type&gt;</code> &ndash; Required. The vttablet's role. Valid values are:
```
Comment (ajm188) [inline content/en/docs/13.0/reference/programs/vtctl/tablets.md]: This isn't correct. `SetWritable` just takes `<tablet-alias> (true|false)`, whereas `ChangeTabletType` takes the `tablet_type` argument):

- https://github.com/vitessio/vitess/blob/179298bc6f14e92e027ee6ee0cda02aefa91536f/go/cmd/vtctldclient/command/tablets.go#L37-L44
- https://github.com/vitessio/vitess/blob/179298bc6f14e92e027ee6ee0cda02aefa91536f/go/cmd/vtctldclient/command/tablets.go#L139-L142

**Review by ajm188 — CHANGES_REQUESTED** [review]

**Comment by ajm188** [comment]
> Before we push this documentation out, I was checking and it looks like this command is not available for vtctlclient. There is no SetWritable action.

The legacy code uses `SetReadOnly` and `SetReadWrite`. When I migrated those, I combined them into `SetWritable` with a true/false argument.

**Comment by FancyFane** [comment]
> > Before we push this documentation out, I was checking and it looks like this command is not available for vtctlclient. There is no SetWritable action.
> 
> The legacy code uses `SetReadOnly` and `SetReadWrite`. When I migrated those, I combined them into `SetWritable` with a true/false argument.

I can see the code in place, however when I build out the current binaries, I'm still seeing the legacy arguments instead of the `SetWritable` argument.

**Comment by kristee-planetscale** [comment]
@ajm188 Updated per your review. Please let me know if there is anything else.

**Comment by ajm188** [comment]
> I can see the code in place, however when I build out the current binaries, I'm still seeing the legacy arguments instead of the SetWritable argument.

`vtctldclient help 2>&1 | egrep '^Set'` (everything that `vtctldclient` can do is accessible (and preferred) via `vtctl VtctldCommand <command>`)

**Review by ajm188 — APPROVED** [review]

**Review by FancyFane — COMMENTED** [review]
As written these commands will fail and will show legacy options:

```sh
jason@melvin:~$ vtctl -h | egrep 'SetRea|SetWrit'
  SetReadOnly <tablet alias>
  SetReadWrite <tablet alias>
```

In order to see the `SetWritable` option you have to use a undocumented `VtctldCommand` and topoology flags first:

```sh
jason@melvin:~$ vtctl ${TOPOLOGY} VtctldCommand -h 2>&1 | egrep 'SetRea|SetWrit'
  SetWritable                Sets the specified tablet as writable or read-only.
```

**Comment by deepthi** [comment]
We now have two sets of commands.
Set 1: vtctl and vtctlclient
Set 2: vtctldclient

For right now, what we should probably do is add `VtctldCommand` to the vtctl reference, and list its subcommands.

I'm not sure whether to create a new page for `vtctldclient` right now, or wait until we reach parity and then replace the existing vtctl docs with the new information.
Opinions welcome :)

**Comment by ajm188** [comment]
> I'm not sure whether to create a new page for vtctldclient right now, or wait until we reach parity and then replace the existing vtctl docs with the new information.
Opinions welcome :)

My two cents:

1. make a new page for `vtctldclient`
2. add documentation for `vtctl VtctldCommand` that includes something to the effect of "this puts you into the same command mode supported by <link-to>vtctldclient</link-to>" 
3. when we reach parity, delete the `vtctlclient` (no "d") page

**Comment by deepthi** [comment]
All vtctldclient commands are now documented. e.g. https://vitess.io/docs/15.0/reference/programs/vtctldclient/vtctldclient_setwritable/
We can close this PR out.

**Pull request closed without merging**